pkg/collector/stats/utils.go

```diff
@@ -224,8 +224,16 @@ func getS390xArchitecture() (string, error) {
 func getCPUMicroArchitecture(family, model, stepping string) (string, error) {
 	yamlBytes, err := os.ReadFile(CPUModelDataPath)
 	if err != nil {
-		klog.Errorf("failed to read cpus.yaml: %v", err)
-		return "", err
+		currentFilePath, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+		LocalCPUModelDataPath := strings.SplitAfter(currentFilePath, "kepler")[0]
+		yamlBytes, err = os.ReadFile(LocalCPUModelDataPath + "/data/cpus.yaml")	
+		if err != nil {
+			klog.Errorf("failed to read cpus.yaml: %v", err)
```

Comment:
```suggestion
```
It's better to return the error to the caller without logging (this was an issue in the existing code). That way the caller can choose to log it, or not.